tools/js-sdk-release-tools/src/common/codeownersAndignorelink/generateCodeOwnersAndIgnoreLink.ts

```diff
@@ -0,0 +1,98 @@
+            options.typeSpecDirectory,
+            options.typespecProject!,
+        );
+        const packageDirectory = await getGeneratedPackageDirectory(
+            typeSpecDirectory,
+            options.sdkRepo.replaceAll("\\", "/"),
+        );
+        const relativePackageDirToSdkRoot = posix.relative(
+            posix.normalize(options.sdkRepo.replaceAll("\\", "/")),
+            posix.normalize(packageDirectory),
+        );
+        await generateCodeOwnersAndIgnoreLinkForPackage(
+            relativePackageDirToSdkRoot,
+        );
+    } else {
+        // Only for hlc.
+        for (const packageFolderPath of options.changedPackagePaths) {
+            logger.info(`packageFolderPath '${packageFolderPath}'`);
+            await generateCodeOwnersAndIgnoreLinkForPackage(packageFolderPath);
+        }
+    }    
+}
+export async function generateCodeOwnersAndIgnoreLinkForPackage(
+    packageFolderPath: string,
+) {
+    logger.info(
+        `Start to generate CODEOWNERS and ignore link for ${packageFolderPath}`,
+    );
+    const jsSdkRepoPath = String(shell.pwd());
+    const packageAbsolutePath = path.join(jsSdkRepoPath, packageFolderPath);
+    const packageName = getNpmPackageName(packageAbsolutePath);
+    const npmViewResult = await tryGetNpmView(packageName);
+    logger.info(`npmViewResult: ${npmViewResult}`);
+    if (!npmViewResult) {
+        logger.info(
+            `Package ${packageName} is first beta release, start to generate CODEOWNERS and ignore link for first beta release.`,
+        );
+        updateCODEOWNERS(packageFolderPath);
+        updateIgnoreLink(packageName);
+        logger.info(
+            `Generated updates for CODEOWNERS and ignore link successfully`,
+        );
+    }
+}
+function updateCODEOWNERS(packagePath: string) {
+    const jsSdkRepoPath = String(shell.pwd());
+    const codeownersPath = path.join(jsSdkRepoPath, ".github", "CODEOWNERS");
+    let content = fs.readFileSync(codeownersPath, "utf8");
+
+    // Insert content before Config section
+    const configSectionIndex = content.indexOf(
+        "###########\n# Config\n###########",
+    );
+    if (configSectionIndex !== -1) {
+        const newContentBeforeConfig = `# PRLabel: %Mgmt\n${packagePath}/ @qiaozha @MaryGao\n`;
+        if (!content.includes(newContentBeforeConfig)) {
+            content =
+                content.slice(0, configSectionIndex) +
+                newContentBeforeConfig +
+                "\n" +
+                content.slice(configSectionIndex);
+        }
+    }
+    fs.writeFileSync(codeownersPath, content);
+}
+
+function updateIgnoreLink(packageName: string) {
+    const jsSdkRepoPath = String(shell.pwd());
+    const ignoreLinksPath = path.join(jsSdkRepoPath, "eng", "ignore-links.txt");
+    let content = fs.readFileSync(ignoreLinksPath, "utf8");
+    const newLine = `https://learn.microsoft.com/javascript/api/${packageName}?view=azure-node-preview`;
```

Comment:
to make sure I understand... Is the issue that the CI has a job that checks that all packages have documentation published?
If that is the case, is there another job that is removing the link from the ignore-links.txt once the documentation is available? 

If my assumption is true, the CI check should be the one that looks if the package is new beta or not and according to that ignore it. No need to do it 1 by 1 and then forgetting to clean up and getting into a place where every sdk documentation link is going to be `ignored` or that there is a discrepancy and our team has to debug to figure out why it is not in sync

Reply:
@maririos the `options.changedPackagePaths` is actually the files in the same package (in line 40), which is misleading by the old script. So this PR should add the package folder instead adding all files' paths to code owner.

There's no another job to remove the ignored link, but is in progress.
This PR checks is first release or not in line 55

Reply:
@wanlwanl I didn't follow. This function is to add the URL of the `learn.microsoft` documentation to the ignore links file.. I understand this applies only to first beta release, as the document generation and publication happens after the SDK is released.

Reply:
There's no another job to remove the ignored link, but is in progress: #10757
Currently, it's a manual work to remove the link in the ignore list.